### PR TITLE
Adding a method to get DNS from device

### PR DESCRIPTION
**Pull request by Lymphatus**
The `getDNS()` method asks the device fro DNS resolvers, both for mobile and WiFi. If it fails for some reason, the app will just use the Google DNS instead.

Tested on API 22 (device) and 23 (emulator).

Also changed the compile SDK and target to API 23 and build-tools to 23.0.1.

### CODE REVIEW

**Comment by Lymphatus** [comment]
Also, shouldn't the app get DNS every time it runs a test, or some of them? They may change since the start (e.g. switching to mobile/WiFi or moving around).

**Comment by bassosimone** [comment]
@Lymphatus, I agree in using the most recent DNS gathered just before the test runs. Since this is also an issue for iOS, I think the best approach is to fix this at the MK level. I've created an issue for this in MK repository: measurement-kit/measurement-kit#331.

File: app/src/main/java/io/github/measurement_kit/app/MainActivity.java [hunk 1]
```diff
@@ -153,5 +161,31 @@ private void copyResources() {
...
+            Method method = SystemProperties.getMethod("get", String.class);
+
+            for (String name : new String[]{"net.dns1", "net.dns2", "net.dns3", "net.dns4",}) {
+                String value = (String) method.invoke(null, name);
+                if (value != null && !value.equals("") && !servers.contains(value)) {
+                    servers.add(value);
+                }
+            }
+        // Using 4 branches to show which errors may occur
+        // We can just catch Exception
```
Comment (bassosimone) [inline app/src/main/java/io/github/measurement_kit/app/MainActivity.java]: I really like this that all the possible errors are enumerated!

**Comment by bassosimone** [comment]
Closes #8.

**Comment by bassosimone** [comment]
Tested, thank you! :+1:

**Comment by bassosimone** [comment]
Actually it closes #4.

**Pull request merged**